jfr-streaming/src/main/java/io/opentelemetry/contrib/jfr/metrics/AbstractThreadDispatchingHandler.java

```diff
@@ -0,0 +1,40 @@
+package io.opentelemetry.contrib.jfr.metrics;
+
+import jdk.jfr.consumer.RecordedEvent;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public abstract class AbstractThreadDispatchingHandler implements RecordedEventHandler {
+  // Will need pruning code for fast-cycling thread frameworks to prevent memory leaks
+  protected final Map<String, RecordedEventHandler> perThread = new HashMap<>();
+  protected final ThreadGrouper grouper;
+
+  public AbstractThreadDispatchingHandler(ThreadGrouper grouper) {
+    this.grouper = grouper;
+  }
+
+  public void reset() {
+    perThread.clear();
+  }
+
+  public abstract String getEventName();
+
+  public abstract RecordedEventHandler createPerThreadSummarizer(String threadName);
+
+  @Override
+  public void accept(RecordedEvent ev) {
+    final Optional<String> possibleGroupedThreadName = grouper.groupedName(ev);
+    possibleGroupedThreadName.ifPresent(
```

Comment:
so I understand, if the thread doesn't have a group name, we don't do anything with it?